store/commitment/iavl/tree.go

```diff
@@ -1,28 +1,39 @@
 package iavl
 
 import (
+	"errors"
 	"fmt"
+	"io"
+	"math"
 
 	dbm "github.com/cosmos/cosmos-db"
+	protoio "github.com/cosmos/gogoproto/io"
 	"github.com/cosmos/iavl"
 	ics23 "github.com/cosmos/ics23/go"
 
 	log "cosmossdk.io/log"
 	"cosmossdk.io/store/v2"
+	snapshottypes "cosmossdk.io/store/v2/snapshots/types"
 )
 
 var _ store.Committer = (*IavlTree)(nil)
 
+var _ snapshottypes.CommitSnapshotter = (*IavlTree)(nil)
+
 // IavlTree is a wrapper around iavl.MutableTree.
 type IavlTree struct {
 	tree *iavl.MutableTree
+
+	// storeKey is the identifier of the store.
+	storeKey string
 }
 
 // NewIavlTree creates a new IavlTree instance.
-func NewIavlTree(db dbm.DB, logger log.Logger, cfg *Config) *IavlTree {
+func NewIavlTree(db dbm.DB, logger log.Logger, storeKey string, cfg *Config) *IavlTree {
```

Comment:
i thought there was discussion around abstracting storekey away? does this change the direction of that conversation?

Reply:
I think the expectation is that `dbm.DB` is a PrefixDb (for iavl v0 & v1) so the store key prefix is already baked into that.  I guess storekey is required for the snapshot API? (still reading this PR).

Reply:
`db dbm.DB` is not required or used in iavl v2 (since it's an interface to an underlying generic KV store).  Maybe we should folder this into `Config` if we want to support both multiple iavl versions in store v2?